File: C1.UWP.FlexChart/VB/StockChart/StockChart/ViewModel/ChangeSymbolCommand.vb
Public Class ChangeSymbolCommand
    Implements ICommand
    Public Event CanExecuteChanged As EventHandler Implements ICommand.CanExecuteChanged
    Public Sub RaiseCanExecuteChanged()
        RaiseEvent CanExecuteChanged(Me, New EventArgs())
    End Sub

    Private _viewModel As ChartViewModel
    Public Sub New(viewModel As ChartViewModel)
        _viewModel = viewModel
    End Sub

    Public Function CanExecute(parameter As Object) As Boolean Implements ICommand.CanExecute
        If _viewModel.ChangeSymbolCommandParamter IsNot Nothing Then
            Dim text = _viewModel.ChangeSymbolCommandParamter
            If Not String.IsNullOrEmpty(text) Then
                Dim code = text.Split(New Char() {" "c}).FirstOrDefault()
                If _viewModel.SymbolNames.Keys.Contains(code) Then
                    Return True
                End If
            End If
        End If
        Return False
    End Function

    Public Sub Execute(parameter As Object) Implements ICommand.Execute
        Dim code = parameter.ToString().Split(New Char() {" "}).FirstOrDefault()
        If Not String.IsNullOrEmpty(code) Then
            _viewModel.MainSymbol = code

            _viewModel.ChangeSymbolCommandParamter = String.Empty
        End If


    End Sub
End Class

File: C1.UWP.FlexChart/CS/StockChart/StockChart/ViewModel/ChangeSymbolCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StockChart.ViewModel
{
    public class ChangeSymbolCommand : System.Windows.Input.ICommand
    {
        public event EventHandler CanExecuteChanged;

        public void RaiseCanExecuteChanged()
        {
            if (this.CanExecuteChanged != null)
            {
                this.CanExecuteChanged(this, new EventArgs());
            }
        }

        ChartViewModel _viewModel;
        public ChangeSymbolCommand(ChartViewModel viewModel)
        {
            _viewModel = viewModel;
        }

        public bool CanExecute(object parameter)
        {
            if (_viewModel.ChangeSymbolCommandParamter != null)
            {
                var text = _viewModel.ChangeSymbolCommandParamter;
                if (!string.IsNullOrEmpty(text))
                {
                    var code = text.Split(new char[] { ' ' }).FirstOrDefault();
                    if (_viewModel.SymbolNames.Keys.Contains(code))
                    {
                        return true;
                    }
                }
            }
            return false;
        }

        public void Execute(object parameter)
        {
            var code = parameter.ToString().Split(new char[] { ' ' }).FirstOrDefault();
            if (!string.IsNullOrEmpty(code))
            {
                _viewModel.MainSymbol = code;

                _viewModel.ChangeSymbolCommandParamter = string.Empty;
            } 


        }
    }
}
